torch/csrc/jit/passes/onnx/constant_map.cpp

```diff
@@ -0,0 +1,177 @@
+#include <torch/csrc/jit/passes/onnx/constant_map.h>
+
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+// Meyer’s Singleton for C++ 14
+ConstantValueMap& ConstantValueMap::getInstance() {
+  static ConstantValueMap s;
+  return s;
+}
+
+void ConstantValueMap::SetRank(
+    const std::string& tensorName,
+    size_t rankValue) {
+  ConstantValueMap::getInstance().rankMap.emplace(tensorName, rankValue);
+}
+
+bool ConstantValueMap::HasRank(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().rankMap.find(tensorName) !=
+      ConstantValueMap::getInstance().rankMap.end();
+}
+
+c10::optional<size_t> ConstantValueMap::GetRank(const std::string& tensorName) {
+  if (!HasRank(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().rankMap[tensorName];
+}
+
+void ConstantValueMap::SetShape(
+    const std::string& tensorName,
+    const c10::SymbolicShape& shapeValue) {
+  ConstantValueMap::getInstance().shapeMap.emplace(tensorName, shapeValue);
+}
+
+bool ConstantValueMap::HasShape(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().shapeMap.find(tensorName) !=
+      ConstantValueMap::getInstance().shapeMap.end();
+}
+
+c10::optional<c10::SymbolicShape> ConstantValueMap::GetShape(
+    const std::string& tensorName) {
+  if (!HasShape(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().shapeMap[tensorName];
+}
+
+void ConstantValueMap::SetValue(
+    const std::string& tensorName,
+    const at::Tensor& value) {
+  ConstantValueMap::getInstance().tensorValueMap.emplace(tensorName, value);
+}
+
+bool ConstantValueMap::HasValue(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().tensorValueMap.find(tensorName) !=
+      ConstantValueMap::getInstance().tensorValueMap.end();
+}
+
+c10::optional<at::Tensor> ConstantValueMap::GetValue(
+    const std::string& tensorName) {
+  if (!HasValue(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().tensorValueMap[tensorName];
+}
+
+c10::optional<std::vector<int64_t>> ConstantValueMap::GetShapeInto1DInt64Vector(
+    const std::string& value_name) {
+  if (ConstantValueMap::HasShape(value_name)) {
+    auto shape_size = ConstantValueMap::GetShape(value_name).value();
+    std::vector<int64_t> shape_value;
+    if (shape_size.isComplete()) {
+      auto shape_symbol_list = shape_size.sizes().value();
+      for (const auto& v : shape_symbol_list) {
+        shape_value.emplace_back(v.static_size());
+      }
+      return c10::optional<std::vector<int64_t>>(shape_value);
```

Comment:
nit: is it necessary to wrap `shape_value` with type cast? i thought it can be done implicitly ..

Reply:
You are right, changed